iosApp/iosApp/Pages/StopDetails/StopDetailsPage.swift

```diff
@@ -80,22 +80,27 @@ struct StopDetailsPage: View {
                     )
                 }
             }
-            .onChange(of: stopDetailsVM.global) { _ in updateDepartures() }
-            .onChange(of: stopDetailsVM.pinnedRoutes) { _ in updateDepartures() }
-            .onChange(of: stopFilter) { _ in updateDepartures() }
+            .onChange(of: stopDetailsVM.global) { _ in updateDepartures(stopFilter: stopFilter) }
+            .onChange(of: stopDetailsVM.pinnedRoutes) { _ in updateDepartures(stopFilter: stopFilter) }
+            .onChange(of: stopFilter) { newStopFilter in
+                if newStopFilter == nil {
+                    internalRouteCardData = nil
+                }
+                updateDepartures(stopFilter: newStopFilter)
+            }
             .onChange(of: stopDetailsVM.stopData) { stopData in
                 errorBannerVM.loadingWhenPredictionsStale = !(stopData?.predictionsLoaded ?? true)
-                updateDepartures()
+                updateDepartures(stopFilter: stopFilter)
             }
             .onChange(of: filters) { nextFilters in setTripFilter(filters: nextFilters) }
             .onChange(of: internalRouteCardData) { _ in
                 let nextStopFilter = setStopFilter()
                 setTripFilter(filters: .init(stopId: stopId, stopFilter: nextStopFilter, tripFilter: tripFilter))
             }
-            .task(id: stopId) {
+            .task(id: "\(stopId) \(stopFilter) \(tripFilter)") {
```

Comment:
planning to make this more like NearbyTransitView and have the task only set now and an onChange that takes all the necessary params for updateDepartures instead of this approach